fe1-web/src/core/network/ingestion/ExtendedMessage.ts

```diff
@@ -41,45 +41,51 @@ export class ExtendedMessage extends Message implements ProcessableMessage {
 
   public readonly receivedFrom: string;
 
-  public processedAt?: Timestamp;
+  public readonly processedAt?: Timestamp;
 
-  // The channel on which the message was received
-  public channel: Channel;
+  public readonly channel: Channel;
+
+  get laoId(): Hash | undefined {
+    try {
+      return getLaoIdFromChannel(this.channel);
+    } catch (e) {
+      return undefined;
+    }
+  }
 
   constructor(msg: Partial<ExtendedMessage>) {
-    super(msg);
+    if (!msg.channel) {
+      throw new Error(
+        "Undefined 'channel' parameter encountered during 'ExtendedMessage' creation",
+      );
+    }
+
+    super(msg, msg.channel);
 
     if (!msg.receivedFrom) {
       throw new Error(
-        '"receivedFrom" is not defined when creating a new instance of ExtendedMessage',
+        "Undefined 'receivedFrom' parameter encountered during 'ExtendedMessage' creation",
       );
     }
+    this.receivedFrom = msg.receivedFrom;
+
+    this.channel = msg.channel;
 
     this.receivedAt = msg.receivedAt || Timestamp.EpochNow();
-    this.receivedFrom = msg.receivedFrom;
     this.processedAt = msg.processedAt;
-    this.channel =
-      msg.channel ||
-      (() => {
-        throw new Error('channel not defined');
-      })();
```

Comment:
Yes but I had to move it before the `super` call since the `Message` constructor also requires it (but doesn't store the value) :)

https://github.com/dedis/popstellar/blob/work-fe1-nico-separate-rollcall-feature/fe1-web/src/core/network/ingestion/ExtendedMessage.ts#L57-L63